csp_billing_adapter/bill_utils.py

```diff
@@ -123,7 +200,44 @@ def process_metering(
     cache: dict,
     hook,
     empty_metering: bool = False
-):
+) -> None:
+    """
+    Handle the CSP metering process, updating the csp_config and cache
+    data stores appropriately.
+    The metering process consists of a number of steps, starting with
+    retrieving the lastest billable usage details, or zeroed usage
+    details if 'empty_metering' is specified as True.
+    Next the billing dimensions hash is determined for those billable
+    usage details, which are used to perform a meter_billing() operation
+    against the CSP, via the registered Pluggy hook.
+    If the meter_billing() operation fails, the csp_config data store will
+    be updated to reflect that failure.
+    Otherwise the csp_config data store will be updated to reflect that
+    a metering operation was successfully performed at that time, with
+    the validity expiration time updated to reflect the next metering
+    time. Additionally, if 'empty_metering' was specified as True, the
```

Comment:
This would be the opposite. If empty metering is False then metering with actual usage is performed and updated in cache. Otherwise only csp config is updated.